src/hyperion/exceptions.py

```diff
@@ -42,6 +42,5 @@ def warn_if_exception_matches(exception: Exception):
         yield from contingency_wrapper(
             func(*args, **kwargs),
             except_plan=warn_if_exception_matches,
-            auto_raise=True,
```

Comment:
Nit: This defaults to `True` so you still have the redundancy that you had before. Though I'm not too bothered as this protects us against future changes in the default.

Reply:
I will take it out. I also just realise that `warn_if_exception_matches` is a plan so it needs a `yield from null()`